### PR TITLE
doc: review 802154 SL

### PR DESCRIPTION
**Pull request by greg-fer**
Reviewed documentation of the 802154 SL lib.

Signed-off-by: Grzegorz Ferenc <grzegorz.ferenc@nordicsemi.no>

### CODE REVIEW

File: doc/links.txt [hunk 1]
```diff
@@ -1,33 +1,39 @@
...
 .. _`LTE Link Monitor`: https://infocenter.nordicsemi.com/topic/ug_link_monitor/UG/link_monitor/lm_certificate_manager.html
 
+.. _`nRF IEEE 802.15.4 radio driver documentation`: https://infocenter.nordicsemi.com/topic/struct_drivers/struct/radio_driver_latest.html
+
+.. ### Links to Nordic pages
+
 .. _`nRF9160 modem firmware zip file`: https://www.nordicsemi.com/Products/Low-power-cellular-IoT/nRF9160/Download#98C9E2578566420786ABD40B695FDB9B
 
-.. _`mbed TLS backend`: https://tls.mbed.org/
+.. ### Links to nrf (not a good way to link)
```
Comment (ankuns) [inline doc/links.txt:17]: `(not a good way to link)` ?

File: doc/links.txt [hunk 2]
```diff
@@ -1,33 +1,39 @@
...
-.. _`Beej's Guide to Network Programming`: https://beej.us/guide/bgnet/
+.. ### Links to Infocenter
 
 .. _`AT Commands Reference Guide`: https://infocenter.nordicsemi.com/topic/ref_at_commands/REF/at_commands/intro.html
 
 .. _`Credential storage management %CMNG`: https://infocenter.nordicsemi.com/topic/ref_at_commands/REF/at_commands/mob_termination_ctrl_status/cmng.html
 
 .. _`LTE Link Monitor`: https://infocenter.nordicsemi.com/topic/ug_link_monitor/UG/link_monitor/lm_certificate_manager.html
 
+.. _`nRF IEEE 802.15.4 radio driver documentation`: https://infocenter.nordicsemi.com/topic/struct_drivers/struct/radio_driver_latest.html
```
Comment (ankuns) [inline doc/links.txt:11]: This link is going to be outdated.
When 1.3.0 relase is out we will have Radio Driver 2.0.0-1.prealpha released there. The doc should point to 2.0.x version. We will be commiting new doc as the Radio Driver evolves to 2.1.0 2.2.0, but in version 1.3.0 of nRF Connect SDK link will not point to 2.0.0-1.prealpha anymore as released but to newer releases which are not part of 1.3.0 nRF Connect SDK.

File: doc/links.txt (hunk 1 above)
Comment (greg-fer) [reply doc/links.txt:17]: Correct, links with `../../..` are not optimal. This comes from the rebase.

File: doc/links.txt (hunk 2 above)
Comment (greg-fer) [reply doc/links.txt]: Updated link to v2.0.0-1.prealpha. We can discuss link versioning after NCS 1.3 release.

**Pull request merged**